### ISSUE TITLE
Sales Rule extensibility problem for actions.

### ISSUE THREAD

**Issue by sergei-sss** [magento/magento2#18359]
### Summary
Magento 2.2
I need to use the default magento conditions from sales rule extension for own entity.
I created own form with identity "mageworx_rewardpoints_rule" instead "sales_rule_form" and extend
blocks for conditions and actions tabs:

```php
namespace MageWorx\RewardPoints\Block\Adminhtml\Reward\Rule\Edit\Tab;

class Actions extends \Magento\SalesRule\Block\Adminhtml\Promo\Quote\Edit\Tab\Actions
{
    /**
     * {@inheritdoc}
     */
    protected function _prepareForm()
    {
        $model = $this->_coreRegistry->registry(\MageWorx\RewardPoints\Model\RegistryConstants::CURRENT_REWARD_RULE);
        $form  = $this->addTabToForm($model, 'actions_fieldset', 'mageworx_rewardpoints_rule_form');
        $this->setForm($form);

        return $this;
    }
}
```

It is possible to save my entity's conditions successfully but I can't save actions, because you use different logic in your controllers.

### Examples

For actions:
correct url: http://prntscr.com/l1pxk7
incorrect response: http://prntscr.com/l1pybi
For conditions I didn't detect the similar issue.

### Proposed solution

Probably, you should add to 
\Magento\SalesRule\Controller\Adminhtml\Promo\Quote\NewActionHtml 
line, as in  \Magento\SalesRule\Controller\Adminhtml\Promo\Quote\NewConditionHtml :

```php
$model->setFormName($formName);
```
in order that the developers can avoid the duplicate code (own NewActionHtml controller and own block for change controller URL).

**Comment by magento-engcom-team** [comment]
Hi @sergei-sss. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@sergei-sss do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [x] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [x] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [x] 6. Add label `Issue: Confirmed` once verification is complete. 

- [x] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by magento-engcom-team** [comment]
@engcom-backlog-nazar Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-96899`, `MAGETWO-96900` were created

**Comment by magento-engcom-team** [comment]
Hi @sergei-sss. 

Thank you for your report and collaboration!

The related internal Jira ticket `MC-24619` was closed as `Fixed`. 

The fix will be available with the upcoming `2.4.3` release.
